### PR TITLE
Ensure we send Apply open email to new candidates

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

I looked at this as i'm adding another email for Find and wanted to shamelessly steal the query. While doing that i noticed that currently, we won't send an email to candidates who have signed up after the apply1 deadline. 

We would also send many emails to a candidate who had many applications so i think we should be distinct'ing the candidate.

This adds some logic to check for applicants with an application form in the current cycle and distinct's candidates.

## Changes proposed in this pull request

- Ensure candidates who signed up post Apply1 deadline are informed of Apply reopening
- Ensure candidates only get 1 email

## Guidance to review

Assuming it makes sense for us to email these candidates too right?

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/queries/get_unsuccessful_and_unsubmitted_candidates.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class GetUnsuccessfulAndUnsubmittedCandidates
+  def self.call
+    Candidate
+    .joins(:application_forms)
+    .where(
+      application_forms: { recruitment_cycle_year: RecruitmentCycle.previous_year, id: ApplicationChoice.where(status: ApplicationStateChange::UNSUCCESSFUL_END_STATES).select(:application_form_id) },
```
Comment (stevehook) [inline app/queries/get_unsuccessful_and_unsubmitted_candidates.rb:6]: I think the conditions here are going to match on any application form that has at least one unsuccessful application choice. I am guessing that you really want application forms that have no successful or pending application choices?

Comment (stevehook) [reply app/queries/get_unsuccessful_and_unsubmitted_candidates.rb:6]: (just realised this was existing code from the renamed query).

**Review by stevehook — COMMENTED** [review]
I've got one question about the query used to find unsuccessful applications (which was not introduced in this PR).

👍 for de-duping the query results. Got me wondering whether there was anything could cause the worker to run twice and send a whole new batch of emails to everyone. I don't think so unless the worker fails halfway through and then retries. I think the only way to mitigate this would be to check with `ChaserSent` because we do write a record here to denote that the email was sent for the given application form.

File: app/queries/get_unsuccessful_and_unsubmitted_candidates.rb (hunk 1 above)
Comment (davidgisbey) [reply app/queries/get_unsuccessful_and_unsubmitted_candidates.rb:6]: Oh good point! Will tweak the tests and see if i can get something working.

Comment (davidgisbey) [reply app/queries/get_unsuccessful_and_unsubmitted_candidates.rb:6]: @stevehook i think this is working now. Thanks for the feedback!

**Review by stevehook — APPROVED** [review]
Tried the new query out locally and I think it works now. (It's not picking up any application forms that have any successful choices.)

**Pull request merged**